test/e2e/kappcontroller/app_status_test.go

```diff
@@ -124,6 +125,117 @@ spec:
 		// template
 		cr.Status.Template.UpdatedAt = metav1.Time{}
 		cr.Status.Template.Stderr = ""
+
+	}
+
+	require.Equal(t, expectedStatus, cr.Status)
+}
+
+func Test_AppStatus_Displays_Deploy_Status(t *testing.T) {
+	env := e2e.BuildEnv(t)
+	logger := e2e.Logger{}
+	kapp := e2e.Kapp{t, env.Namespace, logger}
+	sas := e2e.ServiceAccounts{env.Namespace}
+
+	name := "app-deploy"
+	appYaml := fmt.Sprintf(`
+---
+apiVersion: kappctrl.k14s.io/v1alpha1
+kind: App
+metadata:
+  name: %s
+  annotations:
+    kapp.k14s.io/change-group: kappctrl-e2e.k14s.io/apps
+spec:
+  serviceAccountName: kappctrl-e2e-ns-sa
+  fetch:
+    - inline:
+        paths:
+          file.yml: |
+            apiVersion: v1
+            kind: ConfigMap
+            metadata:
+              name: configmap
+            data:
+              key: value
+  template:
+    - ytt: {}
+  deploy:
+    - kapp:
+        intoNs: %s`, name, env.Namespace) + sas.ForNamespaceYAML()
+
+	cleanUpApp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+	}
+
+	cleanUpApp()
+	defer cleanUpApp()
+
+	logger.Section("deploy", func() {
+		kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name}, e2e.RunOpts{StdinReader: strings.NewReader(appYaml)})
+	})
+
+	out := kapp.Run([]string{"inspect", "-a", name, "--raw", "--tty=false", "--filter-kind=App"})
+	assert.Greater(t, len(out), 1000) // the output yaml should be non-trivial (observed len ~2.7k)
+
+	var cr v1alpha1.App
+	err := yaml.Unmarshal([]byte(out), &cr)
+	if err != nil {
+		t.Fatalf("Failed to unmarshal: %s", err)
+	}
+
+	expectedAppLabel := kapp.Run([]string{"label", "-a", name + ".app", "--tty=false"})
+
+	expectedStatus := v1alpha1.AppStatus{
+		GenericStatus: v1alpha1.GenericStatus{
+			Conditions: []v1alpha1.Condition{{
+				Type:    v1alpha1.ReconcileSucceeded,
+				Status:  corev1.ConditionTrue,
+				Message: "",
+			}},
+			ObservedGeneration:  1,
+			FriendlyDescription: "Reconcile succeeded",
+		},
+		Deploy: &v1alpha1.AppStatusDeploy{
+			ExitCode: 0,
+			Finished: true,
+			Error:    "",
+			KappDeployStatus: v1alpha1.KappDeployStatus{
+				AssociatedResources: v1alpha1.AssociatedResources{
+					Label:      expectedAppLabel,
+					Namespaces: []string{"kappctrl-test"},
+					GroupKinds: []metav1.GroupKind{
+						{Group: "", Kind: "ConfigMap"},
+					},
+				},
+			},
+		},
```

Comment:
since this test specifically focues on kapp deploy status, let's simplify assertion to only care about status.Deploy section or even just status.Deploy.KappDeployStatus

Reply:
yup! done